cilium/cmd/debuginfo.go

```diff
@@ -93,43 +180,85 @@ func runDebugInfo(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	// define output type and file path
 	var output outputType
-	var path string
-
-	switch {
-	case len(file) > 0: // Markdown file
-		output = MARKDOWN
-		path = file
-	case len(html) > 0: // HTML file
-		output = HTML
-		path = html
-	default: // Write to standard output
-		output = STDOUT
-	}
 
 	// create tab-writer to fill buffer
 	var buf bytes.Buffer
 	w := tabwriter.NewWriter(&buf, 5, 0, 3, ' ', 0)
 	p := resp.Payload
 
-	// generate multiple files
-	if (len(file) > 0 || len(html) > 0) && filePerCommand {
-		for cmdName, section := range sections {
+	cmdTime := time.Now()
+
+	if outputToFile && len(outputOpts) == 0 {
+		outputOpts = append(outputOpts, markdownOutput)
+	}
+
+	// Dump payload for each output format.
+	for _, outputOpt := range outputOpts {
+		var fileName string
+
+		switch strings.ToUpper(outputOpt) {
+		case markdownOutput:
+			output = MARKDOWN
+		case htmlOutput:
+			output = HTML
+		case jsonOutputDebuginfo:
+			output = JSONOUTPUT
+		case jsonpathOutput:
+			output = JSONPATH
+		default:
+			if re.MatchString(outputOpt) {
+				output = JSONPATH
+			}
+		}
+
+		if outputToFile {
+			fileName = formatFileName(outputDir, cmdTime, output)
+		}
+
+		// Generate multiple files for each subsection of the command if
+		// specified, except in the JSON cases.
+		if filePerCommand && (output != JSONOUTPUT && output != JSONPATH) {
+			for cmdName, section := range sections {
+				addHeader(w)
+				section(w, p)
+				writeToOutput(buf, output, fileName, cmdName)
+				buf.Reset()
+			}
+			continue
+		}
+
+		// Generate a single file, except not for JSON; no formatting is
+		// needed.
+		if output != JSONOUTPUT && output != JSONPATH {
 			addHeader(w)
-			section(w, p)
-			writeToOutput(buf, output, path, cmdName)
+			for _, section := range sections {
+				section(w, p)
+			}
+			writeToOutput(buf, output, fileName, "")
+			buf.Reset()
+		} else {
```

Comment:
So this is the JSONOUTPUT / JSONPATH case?

IMO it's a bit easier to read conditional statements if you don't have to do the "not not" mental ballet:

```
if output == JSONOUTPUT || output == JSONPATH {
...
} else {
...
}
```